sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -5425,39 +5424,59 @@ pi_result cuda_piextUSMEnqueueMemcpy2D(pi_queue queue, pi_bool blocking,
       (*event)->start();
     }
 
-    // Determine the direction of Copy using cuPointerGetAttributes
+    // Determine the direction of copy using cuPointerGetAttribute
     // for both the src_ptr and dst_ptr
-    // TODO: Doesn't yet support CU_MEMORYTYPE_UNIFIED
-    CUpointer_attribute attributes = {CU_POINTER_ATTRIBUTE_MEMORY_TYPE};
+    CUDA_MEMCPY2D cpyDesc = {0};
 
     CUmemorytype src_type = static_cast<CUmemorytype>(0);
-    void *src_attribute_values[] = {(void *)(&src_type)};
-    result = PI_CHECK_ERROR(cuPointerGetAttributes(
-        1, &attributes, src_attribute_values, (CUdeviceptr)src_ptr));
-    assert(src_type == CU_MEMORYTYPE_DEVICE || src_type == CU_MEMORYTYPE_HOST);
+    // do not throw if cuPointerGetAttribute returns CUDA_ERROR_INVALID_VALUE
+    // checks with PI_CHECK_ERROR are not suggested
+    CUresult ret = cuPointerGetAttribute(
+        &src_type, CU_POINTER_ATTRIBUTE_MEMORY_TYPE, (CUdeviceptr)src_ptr);
+    assert(
+        (src_type != CU_MEMORYTYPE_ARRAY && src_type != CU_MEMORYTYPE_UNIFIED) &&
+        "ARRAY, UNIFIED types are not supported!");
+
+    // pointer not known to the CUDA subsystem (possibly a system allocated ptr)
+    if (ret == CUDA_ERROR_INVALID_VALUE) {
```

Comment:
Is it safe to check only for two possible error codes from `cuPointerGetAttribute`?